### DISCUSSION TITLE
UI to insert link to other recipe

### DISCUSSION THREAD

**Discussion by bartkummel** [mealie-recipes/mealie#3735] [Feature Request]
### First Check

- [X] I used the GitHub search to find a similar requests and didn't find it.
- [X] Checked the [tasks tagged](https://github.com/mealie-recipes/mealie/issues?q=is%3Aissue+is%3Aopen+label%3Atask+) issues and verified my feature is not covered

### Please provide a concise description of the problem that would be addressed by this feature.

There are are some feature requests already where putting a link to another recipe is suggested as a simple solution. (#1761, #2991) The good news is that this already works. The not-so-good news is that linking to a recipe is not exactly user friendly. For starters, you have to know the URL of the recipe. Then, you need to consider stripping off the host name; especially for a self-hosted solution, the chance that the host will change is not unlikely.

### Please provide a concise description of the feature that would resolve your issue.

It would be nice if there were an easier way to insert a link to a recipe. Preferable a UI where you could search for a recipe, then select it to insert a link to it. That link should then be in a format that makes sure the link will keep working:
- if the hostname and/or scheme (`http` or `https`) of the Mealie server changes;
- if the linked-to recipe is renamed.

### Please consider and list out some caveats or tradeoffs made in your design decision

This solves not all of the things mentioned in the requests I linked to, but it does make linking to other recipes a whole lot more accessible. Of course it would add complexity to the code, as with any added feature. Other than that, I can't think of any downsides.

### Additional Information

- [ ] If this is accepted I'm willing to submit a PR to provide this feature
- [ ] If this is accepted I'm willing to help maintain this feature
- [ ] I'm willing to sponsor/pay a developer to do this work

**Comment by boc-the-git** [comment]
I'm not sure Bart if this is actually a different feature request to the ones you've linked, or just an interpretation on how they might be implemented?
Certainly our solution to such a problem is not in providing hyperlinks 🙈

**Reply by bartkummel** [reply]
My idea was that having an easier way to link recipes would be valuable _anyway_, and would be a first step towards the other two items:
- for adding a recipe as ingredient (#1761), we would need some way to find and select a recipe;
- for a "multi preparation recipe" (#2991), we would need that too.

By implementing the generic "search recipe and insert a hyperlink to it" flow first, we would provide value already, provide a (temporary) solution for both other tickets, _and_ implement a component that we could probably reuse in both other tickets. 

So, yes, in a way this can be seen as an idea on how to implement those two. But it can also be seen as an individual feature request that provides value on its own.

**Reply by sjacobflaherty** [reply]
Just coming in to say this would be fantastic.

**Reply by ChipSkylark37** [reply]
+1 on this. There are a few different threads regarding this request and some variations at that. 

One example is making a pizza.

Do I need to make the dough? That can call out to another recipe, or I can just check it off that it's not needed (if I have already prepped it).

Sauce Recipe or just 'Add Prepared Sauce from Linked Recipe'? 

Then the instructions to make the pizza.

It would be interesting, for Pizza specifically or other recipes that you need to make something in advance to have sort of a question based recipe that prompts you a notification:

"DID YOU ALREADY MAKE THE DOUGH AND SAUCE FROM LINKED RECIPES?"
If Yes - Proceed with step 3
If No - Start with Step 1 & 2 that are to make those recipes.

This also calls back to another discussion / request to add the total collective time to prepare recipes. If I need to make Dough 2 days in advance, and sauce the day beforehand, the Pizza recipe _could_ take 3 days total to prepare, but if I already have those prepped, then the Pizza recipe adjusts to only take 1-2 hours (preheat, prepare, cook, etc).
